src/main/java/org/apache/ibatis/migration/commands/NewCommand.java

```diff
@@ -35,35 +44,87 @@ public void execute(String... params) {
     if (paramsEmpty(params)) {
```

Comment:
Unfortunately, I really had to overhaul this function. In a nutshell, I did the following:

1. Normalized types. At the end of all this, a `Reader` is required so a template can be applied. I did that by functionalizing it and still honoring options and `MIGRATIONS_HOME` 
2. Functionalized the bind paramaters so it would be crystal clear what was going into it. As a bonus, I also include the extra `args` supplied to the commandline so that `new description blab foo baz` could capture `description blab foo baz`. These get passed along and allow the new hook to operate based on commandline args. 
3. I wanted it to be _dead_ obvious when `hook.before` and `hook.after` were being called and that they were **literally** before and after the template creation (which is what the original code did)